cirq/protocols/mixture.py

```diff
@@ -79,6 +95,24 @@ def mixture(
                     "but it returned NotImplemented.".format(type(val)))
 
 
+def has_mixture(val: Any) -> bool:
+    """Returns whether the value has a mixture representation.
+
+    Returns:
+        If `val` has a `_has_mixture_` method and its result is not
+        NotImplemented, that result is returned. Otherwise, if the value
+        has a `_mixture_` method return if that has a non-default value.
+        Returns False if neither function exists.
+    """
+    getter = getattr(val, '_has_mixture_', None)
+    result = NotImplemented if getter is None else getter()
+    if result is not NotImplemented:
```

Comment:
This should fallback to `has_unitary`, right? If a gate has a unitary that's a special case of a mixture.